nilearn/input_data/base_masker.py

```diff
@@ -97,6 +97,11 @@ def filter_and_extract(imgs, extraction_function,
                                 func_memory_level=2,
                                 memory_level=memory_level)(imgs)
 
+    if np.any(np.isnan(region_signals)):
+        ensure_finite = True
+    else:
+        ensure_finite = False
```

Comment:
Ok. I think we can remove this whole block ? Since signals that comes out are already replaced by zeros.